bzt/utils.py

```diff
@@ -255,6 +255,17 @@ def traverse(cls, obj, visitor):
                 visitor(val, idx, obj)
                 cls.traverse(obj[idx], visitor)
 
+    def filter(self, rules):
+        keys = set(self.keys())
+        for key in keys:
+            if key not in rules:
+                del self[key]
+            else:
+                if isinstance(rules.get(key), dict) and isinstance(self.get(key), BetterDict):
+                    self.get(key).filter(rules[key])
+                    if not self.get(key):  # clear empty
```

Comment:
The check in blazemeter.py works only on top level while this one works recursive. I don't want to cleanup too much of the item. Currently this check affects only dicts.